src/utils/BatchManager.js

```diff
@@ -151,22 +151,18 @@ class BatchManager {
       if (!renderFn || typeof renderFn !== 'function') {
         // component not registered
         context.statusCode = 404;
-        context.duration = msSince(start);
-        throw notFound(name);
+        return Promise.reject(notFound(name));
       }
 
-      let response = null;
-
-      // render the component!
-      try {
-        context.html = renderFn(context.props);
-      } catch (e) {
-        response = Promise.reject(e);
-      } finally {
-        context.duration = msSince(start);
-      }
-
-      return response;
+      return Promise.resolve(renderFn(context.props));
+    })
+    .then((html) => {
+      context.html = html;
+      context.duration = msSince(start);
```

Comment:
Since this promise is returned, this should `return html`, or `return context`, or something - otherwise it's a promise for `undefined`.